### PR TITLE
EASY-1938: Create API documentation; Spec new POST method

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
fixes EASY-1938

#### When applied it will
* Create API documentation
* Spec new POST method
* 

#### related pull requests on github
repo                       | PR
-------------------------- | -----------------
easy-                      | [PR#](PRlink)

### CODE REVIEW

**Comment by rvanheest** [comment]
With the introduction of this API, shouldn't the README in the root be modified as well?

File: docs/_config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+title: easy-deposit-api
```
Comment (janvanmansum) [inline docs/_config.yml]:
```suggestion
title: easy-deposit-agreement-creator
```

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+description: >-
+  JSON-based deposit service.
+baseurl: "/easy-deposit-api"
```
Comment (janvanmansum) [inline docs/_config.yml]:
```suggestion
baseurl: "/easy-deposit-agreement-creator"
```

File: docs/_config.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+title: easy-deposit-api
+description: >-
+  JSON-based deposit service.
```
Comment (janvanmansum) [inline docs/_config.yml]: Adjust description.

File: docs/api/api.yml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+          description: OK
+          content:
+            'application/pdf': {}
+        400:
+          $ref: "#/components/responses/MalformedMetadata"
+        500:
+          $ref: "#/components/responses/InternalServerError"
+  /create:
+    post:
+      summary: Returns a Deposit Agreement based on the datasetId
```
Comment (janvanmansum) [inline docs/api/api.yml]:
```suggestion
      summary: Returns a Deposit Agreement based on the EASY datasetId in Fedora
```

File: docs/api/api.yml [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+          type: string
+    # Note: it is not possible to provide examples in combination with a reference to this
+    # schema. If you need examples, you'll have to spell out the complete type.
+    StringArray:
+      type: array
+      items:
+        type: string
+    IncludeSampleHeader:
+      type: boolean
+      description: Whether the resulting Agreement is a sample, or a final Agreement
```
Comment (janvanmansum) [inline docs/api/api.yml]: Is this called a "sample"? That sounds like "specimen" with some random data. I think "draft" would be a better term here.

Comment (lindareijnhoudt) [reply docs/api/api.yml]: i agree, but this is what it is called. i merely described the status quo

**Pull request merged**